README.md

```diff
@@ -1,6 +1,98 @@
-See the documentation through mkdocs:
+# Meta-Data extraction with MetaLookup
+The _MetaLookup_ service extracts and suggests metadata related to OER from a given URL. To achieve this, it
+uses [Splash](https://splash.readthedocs.io/en/stable/) to fetch the content of the URL and then analyzes the
+[HAR](http://www.softwareishard.com/blog/har-12-spec) (cookies & headers) and the received HTML document. It further
+uses [Lighthouse](https://github.com/GoogleChrome/lighthouse) to assess the accessibility of the content.
+It requires iterative optimization and continuous maintenance due to the ever-changing nature of URL content.
 
-In project folder execute in terminal:
+## Feature Documentation
+The documentation which metadata fields are extracted and how can be found [here](./docs/features.md) and
+[here](./docs/acceptance.md). It can be rendered as HTML with the following command:
 ```bash
 mkdocs serve
 ```
+The different extractors are implemented [here](./src/metalookup/features).
+
+## Local Setup
+MetaLookup requires python 3.9 or later and [poetry](https://python-poetry.org/docs/#installation). To test the
+application without mocking external services, [docker](https://www.docker.com/) is required as well.
+
+With poetry installed running the following commands will set up a poetry environment and install application and
+development dependencies.
+```bash
+poetry env use 3.9
+poetry install
+```
+
+## Code Style & Pre-commit hook
+The code style enforced via tools configured in [.pre-commit-config.yaml](./.pre-commit-config.yaml) and
+[pyproject.toml](./pyproject.toml). After setting up your environment (which will install the python
+`pre-commit` script) one should install the hook via:
+```bash
+pre-commit install
+```
+This will make sure your commits will be accepted by the CI Pipeline.
```

Comment:
Why?

Reply:
Because, the CI pipeline validates codestyle, and if you check in stuff that violates the codestyle the Pipeline will already fail before the unittests

Reply:
added clarification